[CI] RankDocsRetrieverBuilderTests testRewrite failing

**Build Scans:**
- [elasticsearch-intake #10984 / part1](https://gradle-enterprise.elastic.co/s/bgbxloc3issws)
- [elasticsearch-periodic #4367 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/lcfjq2e2sbzck)
- [elasticsearch-periodic #4367 / openjdk17_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/4legg4rfnzriw)
- [elasticsearch-periodic-platform-support #4341 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/m7rg47rjztr3k)
- [elasticsearch-intake #10972 / part1](https://gradle-enterprise.elastic.co/s/7ojv5aqvuokx4)
- [elasticsearch-intake #10966 / part1](https://gradle-enterprise.elastic.co/s/mrtemgpyqii7q)
- [elasticsearch-pull-request #35635 / part-1](https://gradle-enterprise.elastic.co/s/5hm7sbb3w34dq)
- [elasticsearch-pull-request #35609 / part-1](https://gradle-enterprise.elastic.co/s/sqhr7g5nlkdoi)
- [elasticsearch-periodic-platform-support #4337 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/5obttyfkmnnra)
- [elasticsearch-periodic #4363 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/gassgaatg3zjq)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.search.retriever.RankDocsRetrieverBuilderTests.testRewrite" -Dtests.seed=33738B6C762422E8 -Dtests.locale=gl-Latn-ES -Dtests.timezone=Asia/Singapore -Druntime.java=22
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testRewrite'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.search.retriever.RankDocsRetrieverBuilderTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: retriever sources should be rewritten first
```

**Issue Reasons:**
- [8.x] 2 consecutive failures in step openjdk17_checkpart1_java-fips-matrix
- [8.x] 13 failures in test testRewrite (8.0% fail rate in 163 executions)
- [8.x] 3 failures in step part1 (10.7% fail rate in 28 executions)
- [8.x] 3 failures in step part-1 (8.3% fail rate in 36 executions)
- [8.x] 2 failures in step openjdk17_checkpart1_java-fips-matrix (66.7% fail rate in 3 executions)
- [8.x] 3 failures in pipeline elasticsearch-intake (10.7% fail rate in 28 executions)
- [8.x] 3 failures in pipeline elasticsearch-pull-request (8.8% fail rate in 34 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic (50.0% fail rate in 4 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic-platform-support (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -248,9 +248,6 @@ tests:
 - class: org.elasticsearch.xpack.inference.InferenceRestIT
   method: test {p0=inference/40_semantic_text_query/Query a field that uses the default ELSER 2 endpoint}
   issue: https://github.com/elastic/elasticsearch/issues/114376
-- class: org.elasticsearch.search.retriever.RankDocsRetrieverBuilderTests
-  method: testRewrite
-  issue: https://github.com/elastic/elasticsearch/issues/114467
 - class: org.elasticsearch.packaging.test.DockerTests
   method: test022InstallPluginsFromLocalArchive
   issue: https://github.com/elastic/elasticsearch/issues/111063

